File: source/240.搜索二维矩阵-ii.cpp
/*
 * @lc app=leetcode.cn id=240 lang=cpp
 *
 * [240] 搜索二维矩阵 II
 */
#include "cpp_includes.h"
// @lc code=start
class Solution {
 public:
  bool searchMatrix(vector<vector<int>>& matrix, int target) {
    int rows = matrix.size() - 1;
    int cols = 0;

    while (rows >= 0 && cols < matrix[0].size()) {
      if (matrix[rows][cols] == target) {
        return true;
      } else if (matrix[rows][cols] > target) {
        rows--;
      } else {
        cols++;
      }
    }

    return false;
  }
};
// @lc code=end

File: source/240.搜索二维矩阵-ii.cs
/*
 * @lc app=leetcode.cn id=240 lang=csharp
 *
 * [240] 搜索二维矩阵 II
 */

// @lc code=start
public class Solution {
    public bool SearchMatrix(int[,] matrix, int target) {
        int m = matrix.GetLength(0);
        int n = matrix.GetLength(1);

        int row = m - 1;
        int col = 0;

        while (row >= 0 && col < n) {
            if (matrix[row, col] == target) return true;

            if (matrix[row, col] > target) {
                row--;
            } else {
                col++;
            }
        }
        return false;
    }
}
// @lc code=end

